lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -865,8 +876,13 @@ SerializedModuleLoaderBase::loadModule(SourceLoc importLoc,
   Ctx.LoadedModules[moduleID.first] = M;
   SWIFT_DEFER { M->setHasResolvedImports(); };
 
-  if (!loadAST(*M, moduleID.second, std::move(moduleInputBuffer),
-               std::move(moduleDocInputBuffer), std::move(moduleSourceInfoInputBuffer),
+  StringRef moduleInterfacePathStr =
+    Ctx.AllocateCopy(StringRef(moduleInterfacePath.c_str(),
+                               moduleInterfacePath.size()));
```

Comment:
There's an AllocateCopy overload for StringRefs already. (Also, good catch that you need to copy the string into long-term storage.)

Reply:
I don't follow... are you asking me to get rid of the explicit `StringRef` constructor? If I do that, how will it call the correct `AllocateCopy` overload?

Reply:
Ah, right, `moduleInterfacePath` is a SmallString, which is also a SmallVectorImpl. Still, `moduleInterfacePath.str()` would be simpler.